### PR TITLE
storage pools: implement rsync.bwlimit property

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

**Comment by brauner** [comment]
I think `buildd06` went down.

**Comment by brauner** [comment]
jenkins: test this please

File: doc/configuration.md [hunk 1]
```diff
@@ -398,6 +398,7 @@ volume.block.mount\_options     | string    | block based driver (lvm)
 lvm.thinpool\_name              | string    | lvm driver                        | LXDPool           | Thin pool where images and containers are created.
 lvm.use\_thinpool               | bool      | lvm driver                        | true              | Whether the storage pool uses a thinpool for logical volumes.
 lvm.vg\_name                    | string    | lvm driver                        | name of the pool  | Name of the volume group to create.
+rsync.bwlimit                   | string    | -				        | 0 (no limit)      | Specifies the upper limit to be placed on the socket I/O whenever rsync has to be used to transfer storage entities.
```
Comment (stgraber) [inline doc/configuration.md]: Bad column alignment

File: doc/storage-backends.md [hunk 2]
```diff
@@ -43,6 +43,9 @@ LXD uses those features to transfer containers and snapshots between servers.
 When such capabilities aren't available, either because the storage driver doesn't support it  
 or because the storage backend of the source and target servers differ,  
 LXD will fallback to using rsync to transfer the individual files instead.
+When rsync has to be used LXD allows to specify an upper limit on the amount of
```
Comment (stgraber) [inline doc/storage-backends.md]: Make that a separate paragraph please, markdown rendering is pretty bad as it is :)

File: lxd/api_1.0.go [hunk 3]
```diff
@@ -102,6 +102,7 @@ func api10Get(d *Daemon, r *http.Request) Response {
 			"storage_zfs_clone_copy",
 			"unix_device_rename",
 			"storage_lvm_use_thinpool",
+			"storage_rsync_bwlimit",
```
Comment (stgraber) [inline lxd/api_1.0.go]: You're missing the doc/api-extensions.md entry for this.

File: lxd/storage_lvm.go [hunk 4]
```diff
@@ -915,19 +915,20 @@ func (s *storageLvm) StoragePoolUpdate(writable *api.StoragePoolPut, changedConf
 		return fmt.Errorf("The \"zfs.pool_name\" property does not apply to LVM drivers.")
 	}
 
+	if shared.StringInSlice("lvm.use_thinpool", changedConfig) {
+		return fmt.Errorf("The \"lvm.use_thinpool\" property cannot be changed.")
+	}
```
Comment (stgraber) [inline lxd/storage_lvm.go]: What's that doing here?

**Comment by stgraber** [comment]
what's the unit of rsync.bwlimit? bytes?

File: doc/configuration.md (hunk 1 above)
Comment (brauner) [inline doc/configuration.md]: done

File: doc/storage-backends.md (hunk 2 above)
Comment (brauner) [inline doc/storage-backends.md]: done

File: lxd/api_1.0.go (hunk 3 above)
Comment (brauner) [inline lxd/api_1.0.go]: done

**Comment by brauner** [comment]
@stgraber, everything that is parseable by `ParseByteSizedString()`.

**Comment by stgraber** [comment]
Perfect, I was just wondering as the tests use 1024 as a limit which seemed a bit low :)

**Pull request merged**